integration/specs/TimePicker/timePicker-1.spec.js

```diff
@@ -0,0 +1,576 @@
+        picklist.clickDownButton();
+        expect(picklist.getHourValue()).toBe('12');
+    });
+    it('should set hour input value to "01" when hour input has "0" as value and press up button while hour input is focused', () => {
+        const picklist = new PageTimePicker(TIME_PICKER);
+        picklist.click();
+        picklist.waitUntilOpen();
+        picklist.setHourValue('0');
+        picklist.clickUpButton();
+        expect(picklist.getHourValue()).toBe('01');
+    });
+    it('should set hour input value to "01" when press up key while hour input is focused', () => {
+        const picklist = new PageTimePicker(TIME_PICKER);
+        picklist.click();
+        picklist.waitUntilOpen();
+        browser.keys(ARROW_UP_KEY);
+        expect(picklist.getHourValue()).toBe('01');
+    });
+    it('should set hour input value to "01" when hour input has "0" as value and press up key while hour input is focused', () => {
+        const picklist = new PageTimePicker(TIME_PICKER);
+        picklist.click();
+        picklist.waitUntilOpen();
+        picklist.setHourValue('0');
+        browser.keys(ARROW_UP_KEY);
+        expect(picklist.getHourValue()).toBe('01');
+    });
+    it('should set the right hour input value and focus minutes input when type "2" and hour input is focused', () => {
+        const picklist = new PageTimePicker(TIME_PICKER);
+        picklist.click();
+        picklist.waitUntilOpen();
+        browser.keys('2');
+        expect(picklist.hasFocusMinutesInput()).toBe(true);
+        expect(picklist.getHourValue()).toBe('02');
+    });
+    it('should set the right hour input value and focus minutes input when type "02" and hour input is focused', () => {
+        const picklist = new PageTimePicker(TIME_PICKER);
+        picklist.click();
+        picklist.waitUntilOpen();
+        browser.keys('0');
+        browser.keys('2');
+        expect(picklist.hasFocusMinutesInput()).toBe(true);
+        expect(picklist.getHourValue()).toBe('02');
+    });
+    it('should set the right hour input value and focus minutes input when type "15" and hour input is focused', () => {
+        const picklist = new PageTimePicker(TIME_PICKER);
+        picklist.click();
+        picklist.waitUntilOpen();
+        browser.keys('1');
+        browser.keys('5');
+        expect(picklist.hasFocusMinutesInput()).toBe(true);
+        expect(picklist.getHourValue()).toBe('03');
+    });
+    it('should set the right hour input value and keep the focus on hour input when type "0" and hour input is focused', () => {
+        const picklist = new PageTimePicker(TIME_PICKER);
+        picklist.click();
+        picklist.waitUntilOpen();
+        browser.keys('0');
+        expect(picklist.hasFocusHourInput()).toBe(true);
+        expect(picklist.getHourValue()).toBe('00');
+    });
+    it('should set the right hour input value and keep the focus on hour input when type "1" and hour input is focused', () => {
```

Comment:
here when you say:
```keep the focus on hour input```
this implies that the hour input has the focus so there is not need to be redundant saying that:
```and hour input is focused```